stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -288,6 +289,32 @@ public class ChatClient internal constructor(
         return eventsObservable.subscribe(filter, listener)
     }
 
+    public fun subscribeFor(
+        lifecycleOwner: LifecycleOwner,
+        vararg eventTypes: String,
+        listener: (event: ChatEvent) -> Unit
+    ): Disposable {
+        val disposable = subscribeFor(
+            *eventTypes,
+            listener = { event ->
+                if (lifecycleOwner.lifecycle.currentState.isAtLeast(Lifecycle.State.STARTED)) {
+                    listener(event)
+                }
+            }
+        )
+
+        lifecycleOwner.lifecycle.addObserver(
+            object : LifecycleObserver {
+                @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+                fun onDestroy() {
+                    disposable.dispose()
+                }
+            }
+        )
```

Comment:
This performs _most_ of what `LiveData` does for safe observations:

- It only delivers values if your lifecycle is in a started state, so that you don't touch UI when you shouldn't
- It cleans up the observer when the lifecycle ends

It's missing one thing: if the lifecycle is not _started_, it won't queue values and wait for it to become _started_ again - it will simply drop them.

Reply:
I think it is fine, it should be started when it is subscribed